servant/src/Servant/Utils/Enter.hs

```diff
@@ -39,6 +39,9 @@ instance ( Enter typ1 arg1 ret1, Enter typ2 arg2 ret2
 instance (Enter b arg ret) => Enter (a -> b) arg (a -> ret) where
     enter arg f a = enter arg (f a)
 
+instance Enter (arg -> ret) arg ret where
+    enter arg f = f arg
```

Comment:
Yes, I think that's better. I wouldn't include it in `servant` then, though.

@jkarni: @amarpotghan and I tried for some time to come up with a nice general way of mapping over the leafs of a `:<|>`-`:>`-structure, but didn't manage. We hoped that this general mapping function would also allow to implement `staticClient`. Now since it's much less general, I think that this mapping should stay in `servant-client`.

(We could still move `enter` to `servant`, since it's been proven to be useful in e.g. `servant-client`, but that'd be a completely unrelated change. #NoScopeCreep)